packages/kolme-store/src/error.rs

```diff
@@ -1,15 +1,36 @@
 use merkle_map::{MerkleSerialError, Sha256Hash};
 
-#[derive(thiserror::Error, Debug)]
+#[derive(Debug, Clone, Copy, serde::Serialize, serde::Deserialize)]
+pub enum StorageBackend {
+    Fjall,
+    Postgres,
+    InMemory,
+}
+
+impl std::fmt::Display for StorageBackend {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            StorageBackend::Fjall => write!(f, "Fjall"),
+            StorageBackend::Postgres => write!(f, "Postgres"),
+            StorageBackend::InMemory => write!(f, "InMemory"),
+        }
+    }
+}
+
+#[derive(thiserror::Error, Debug, Clone, serde::Serialize, serde::Deserialize)]
 pub enum KolmeStoreError {
-    #[error(transparent)]
-    Custom(Box<dyn std::error::Error + Send + Sync>),
-    #[error(transparent)]
-    Merkle(#[from] MerkleSerialError),
+    #[error("Custom error: {0}")]
+    Custom(String),
+
+    #[error("Merkle error: {0}")]
+    Merkle(String),
```

Comment:
I think this could stay as its previous implementation; making it use String just loses information that used to be there.